pkg/daemon/daemon.go

```diff
@@ -425,19 +442,19 @@ func (dn *Daemon) nodeStateChangeHandler(old, new interface{}) {
 	} else if !ok {
 		if err := dn.annotateNode(dn.name, annoIdle); err != nil {
 			glog.Errorf("drainNode(): failed to annotate node: %v", err)
-			dn.exitCh <- err
-			return
+			return err
 		}
 	}
 	dn.refreshCh <- Message{
 		syncStatus:    "Succeeded",
 		lastSyncError: "",
 	}
+	return nil
 }
 
 func (dn *Daemon) completeDrain() {
 	if err := drain.RunCordonOrUncordon(dn.drainer, dn.node, false); err != nil {
-		glog.Errorf("nodeStateChangeHandler(): fail to uncordon node: %v", err)
+		glog.Errorf("nodeStateSyncHandler(): fail to uncordon node: %v", err)
```

Comment:
Remove the `dn.exitCh` when `annotateNode` or `RunCordonOrUncordon` fail and add a error check after `dn.completeDrain`.

Reply:
Fixed.